### PR TITLE
Document Arm Containers

### LINKED ISSUE #218: Add ARM support for Linux
# Description

The ARM architecture, aarch64 and eventually armhf, are gaining traction since the raise of Raspberry PIs and recently cheap ARM offerings in various clouds.
Several projects now fully supports ARM, thus needing a CI.

# Context

The Crystal Language aims to fully support Linux [aarch64](https://github.com/crystal-lang/crystal/issues/7587), and eventually armhf. That's not easy since the current CI doesn't support this architecture.

# Anything Else

Thanks to have suggested us [to file a feature request in our issue tracker](https://github.com/crystal-lang/crystal/issues/7587#issuecomment-476259742), I don't know how you have found our thread :sweat_smile:

### PR DESCRIPTION
**Pull request by fkorotkov**
TLDR now you can use `arm_container` instead of `container` to run `arm64` containers.

```yaml
task:
  arm_container:
    image: ubuntu:latest
  script: uname -a
```

Fixes #218

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Is there an issue where arm doesn't upload the logs? See for example: https://cirrus-ci.com/task/4712248254922752?logs=clone#L1

**Comment by fkorotkov** [comment]
@MarcoFalke sorry for the inconvenience. We were testing and changing some last things. Now everything should work as expected

**Comment by real-or-random** [comment]
There seem to be some more issues: https://cirrus-ci.com/task/5504774142427136 (Or is this a bug in our config?)

**Comment by MarcoFalke** [comment]
Maybe an intermittent issue? In our repo it still works fine.

**Comment by fkorotkov** [comment]
@MarcoFalke, yeah, `dockerfile` for `arm_container`s is not yet supported.

**Comment by real-or-random** [comment]
Oh okay, I see, thank for letting us know!

**Comment by real-or-random** [comment]
>  `dockerfile` for `arm_container`s is not yet supported. It's related to #900

Are there plans to support this now that #900 has been done?

 Basically all of our current Linux builds use a Docker image. If we could port that easily to ARM, I think this would help us a lot.

**Comment by fkorotkov** [comment]
@real-or-random actually now it's much simpler since GCP has Arm instances added #1027. Adding some parser improvements in https://github.com/cirruslabs/cirrus-cli/pull/586 and shortly after we should be able to add `dockerfile` support. Note that instead of a single multi-arch container we'll build two separate images for each architecture.

**Comment by real-or-random** [comment]
Sounds great!

> Note that instead of a single multi-arch container we'll build two separate images for each architecture.

Yep, and I also believe that's in general the better of the two alternatives.
